src/charm.py

```diff
@@ -116,6 +109,31 @@ def _configure_pebble_layer(
         self._container.replan()
         self.unit.status = ActiveStatus()
 
+    def _on_fiveg_nrf_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Handle fiveg-nrf relation joined event.
+
+        Args:
+            event: RelationJoinedEvent
+        """
+        if self.unit.is_leader():
+            self.nrf_provider.set_nrf_information(
```

Comment:
I'm not 100% sure this is necessary but I used to check that the nrf service was running before providing it to other NF's. I'm not sure what will happen if we send it to apps and the url/port aren't accessible yet.

Ref [here](https://github.com/gruyaume/nrf-operator/blob/main/src/charm.py#L100)

Reply:
I added it, I guess there is no harm in checking anyway :)